### PR TITLE
Implemented quick menu widget.

### PR DESCRIPTION
**Pull request by dkaera**
┆Issue is synchronized with this [Jira Task](https://highvibe.atlassian.net/browse/HVN1-238) by [Unito](https://www.unito.io/learn-more)

### CODE REVIEW

File: mobile/lib/modules/home/widgets/quick_menu_item.dart [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  String text;
+  IconData icon;
+  ImageProvider image;
+  String tooltip;
+  Color textColor;
+  AnimationStyle animationStyle;
+  OnQuickMenuItemPressed onPressed;
+  bool hideOnClick;
+
+  QuickMenuItem.withText(this.icon, this.elevation, this.tooltip,
```
Comment (AlexVegner) [inline mobile/lib/modules/home/widgets/quick_menu_item.dart:19]: A lot of unnesesary constructore. Please update it with one constructor with named arguments
const QuickMenuItem() {
 @required this.elevation, 
  @required this.tooltip,
  @required this.onPressed, 
  this.hideOnClick = true,
  this.image,
  this.icon,
  this.text, 
  this.textColor,
}

File: mobile/lib/modules/home/widgets/quick_menu_item.dart [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+      this.onPressed, this.text, this.textColor, this.hideOnClick);
+
+  QuickMenuItem.withTextWithImage(this.image, this.elevation, this.tooltip,
+      this.onPressed, this.text, this.textColor, this.hideOnClick);
+
+  QuickMenuItem.noText(this.icon, this.elevation, this.tooltip, this.onPressed,
+      this.hideOnClick) {
+    text = null;
```
Comment (AlexVegner) [inline mobile/lib/modules/home/widgets/quick_menu_item.dart:27]: By default all fields are null, you don't need to initialize it with null.

File: mobile/lib/modules/home/widgets/quick_menu_item.dart [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+      this.icon,
+      this.image,
+      this.textColor,
+      this.tooltip,
+      this.index,
+      this.controller,
+      this.onPressed,
+      this.hideWidget,
+      this.animationStyle,
+      this.itemCount})
```
Comment (AlexVegner) [inline mobile/lib/modules/home/widgets/quick_menu_item.dart:53]: Use last coma for good flutter formatting:
```dart
    ...
    this.itemCount,
  },
)
```

File: mobile/lib/modules/home/widgets/quick_menu_widget.dart [hunk 4]
```diff
@@ -0,0 +1,120 @@
+import 'package:flutter/material.dart';
+import 'package:highvibe/modules/home/widgets/animation_style.dart';
+import 'package:highvibe/modules/home/widgets/quick_menu_item.dart';
+
+class QuickMenuWidget extends StatefulWidget {
+  const QuickMenuWidget(this._quickMenuItemList,
```
Comment (AlexVegner) [inline mobile/lib/modules/home/widgets/quick_menu_widget.dart:6]: Use constructor with named parameters for all StatefulWidget / StatelessWidget widgets 
Google prefer to do it for all Flutter widgets.
```dart 
  const QuickMenuWidget({
    Key key,
    @required this.quickMenuItemList,
    this.fabAnimationStyle = AnimationStyle.defaultAnimation,
}) : super(key: key);

```

File: mobile/lib/modules/home/widgets/quick_menu_widget.dart [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+      [this._fabAnimationStyle = AnimationStyle.defaultAnimation]);
+
+  final List<QuickMenuItem> _quickMenuItemList;
```
Comment (AlexVegner) [inline mobile/lib/modules/home/widgets/quick_menu_widget.dart:9]: All class fields should be above constructors. 
Keep order: class fields, constructors, methods.

File: mobile/lib/modules/home/widgets/quick_menu_widget.dart [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+  final AnimationStyle _fabAnimationStyle;
+
+  @override
+  QuickMenuWidgetState createState() =>
+      QuickMenuWidgetState(_quickMenuItemList, _fabAnimationStyle);
```
Comment (AlexVegner) [inline mobile/lib/modules/home/widgets/quick_menu_widget.dart:14]: Avoid to send StatefulWidget fields to State 
Use `widget.quickMenuItemList` and `widget.quickMenuItemList` inside State

File: mobile/lib/modules/home/widgets/quick_menu_widget.dart [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+class QuickMenuWidgetState extends State<QuickMenuWidget>
+    with TickerProviderStateMixin {
+  QuickMenuWidgetState(this._quickMenuItemList, this._fabAnimationStyle);
+
+  int _angle = 90;
+  bool _isRotated = true;
+  final List<QuickMenuItem> _quickMenuItemList;
```
Comment (AlexVegner) [inline mobile/lib/modules/home/widgets/quick_menu_widget.dart:23]: remove it

File: mobile/lib/modules/home/widgets/quick_menu_widget.dart [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+  final AnimationStyle _fabAnimationStyle;
```
Comment (AlexVegner) [inline mobile/lib/modules/home/widgets/quick_menu_widget.dart:24]: remove it

File: mobile/lib/modules/home/widgets/quick_menu_widget.dart [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+  @override
+  void initState() {
+    _controller = AnimationController(
+      vsync: this,
+      duration: const Duration(milliseconds: 180),
+    );
+
+    _controller.reverse();
+
+    setFabMenu(_quickMenuItemList);
```
Comment (AlexVegner) [inline mobile/lib/modules/home/widgets/quick_menu_widget.dart:38]: Use `widget.quickMenuItemList` and `widget.quickMenuItemList` inside State

**Review by AlexVegner — CHANGES_REQUESTED** [review]

File: mobile/lib/modules/home/widgets/quick_menu_widget.dart [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+  bool _isRotated = true;
+  final List<QuickMenuItem> _quickMenuItemList;
+  final AnimationStyle _fabAnimationStyle;
+  List<QuickMenuItemWidget> _fabMenuItems;
+
+  AnimationController _controller;
+
+  @override
+  void initState() {
+    _controller = AnimationController(
```
Comment (AlexVegner) [inline mobile/lib/modules/home/widgets/quick_menu_widget.dart:31]: You need to dispose all controllers. in dispose method

**Review by AlexVegner — CHANGES_REQUESTED** [review]

File: mobile/lib/modules/home/widgets/quick_menu_item.dart [hunk 11]
```diff
@@ -0,0 +1,241 @@
+import 'package:flutter/material.dart';
+import 'package:highvibe/modules/home/widgets/animation_style.dart';
+
+typedef OnQuickMenuItemPressed = void Function();
+
+typedef HideWidget = void Function();
+
+class QuickMenuItem {
+  double elevation;
```
[the remark this answers is not in the document]
Comment (dkaera) [reply mobile/lib/modules/home/widgets/quick_menu_item.dart:9]: applied

File: mobile/lib/modules/home/widgets/quick_menu_item.dart (hunk 1 above)
Comment (dkaera) [reply mobile/lib/modules/home/widgets/quick_menu_item.dart:19]: applied

File: mobile/lib/modules/home/widgets/quick_menu_item.dart (hunk 2 above)
Comment (dkaera) [reply mobile/lib/modules/home/widgets/quick_menu_item.dart:27]: removed as not used

File: mobile/lib/modules/home/widgets/quick_menu_item.dart [hunk 12]
```diff
@@ -0,0 +1,232 @@
...
+        child: SvgPicture.asset(
+          svgAssert,
+          fit: BoxFit.none,
+        ),
+        decoration: const BoxDecoration(
+            shape: BoxShape.circle,
+            gradient: LinearGradient(
+                begin: Alignment.topLeft,
+                end: Alignment.bottomLeft,
+                colors: [Color(0xFF666CCC), Color(0xFF66CCCC)])),
```
Comment (AlexVegner) [inline mobile/lib/modules/home/widgets/quick_menu_item.dart]: Please move all colors and gradient list to AppColors

File: mobile/lib/modules/home/widgets/quick_menu_widget.dart [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+                        child: const Icon(
+                          Icons.add,
+                          size: 25,
+                        )),
+                    decoration: const BoxDecoration(
+                        shape: BoxShape.circle,
+                        gradient: LinearGradient(
+                            begin: Alignment.topLeft,
+                            end: Alignment.bottomLeft,
+                            colors: [Color(0xFF666CCC), Color(0xFF66CCCC)])),
```
Comment (AlexVegner) [inline mobile/lib/modules/home/widgets/quick_menu_widget.dart]: Please move all colors and gradient list to AppColors

File: mobile/lib/modules/home/widgets/quick_menu_item.dart [hunk 14]
```diff
@@ -0,0 +1,232 @@
...
+        style: TextStyle(color: textColor, fontSize: 12.0),
+      ),
+      padding: const EdgeInsets.all(8.0),
+      decoration: const BoxDecoration(
+          borderRadius: BorderRadius.all(Radius.circular(6.0)),
+          shape: BoxShape.rectangle,
+          gradient: LinearGradient(
+              begin: Alignment.topLeft,
+              end: Alignment.bottomLeft,
+              colors: [Color(0xFF666CCC), Color(0xFF66CCCC)])),
```
Comment (AlexVegner) [inline mobile/lib/modules/home/widgets/quick_menu_item.dart]: Please move all colors and gradient list to AppColors

**Review by AlexVegner — APPROVED** [review]

**Pull request merged**